client/question/services/FeedbackGeneratorService.js

```diff
@@ -24,18 +24,34 @@ tie.factory('FeedbackGeneratorService', [
   'RUNTIME_ERROR_FEEDBACK_MESSAGES', 'WRONG_LANGUAGE_ERRORS', 'LANGUAGE_PYTHON',
   'CLASS_NAME_AUXILIARY_CODE', 'CLASS_NAME_SYSTEM_CODE', 'PARAGRAPH_TYPE_TEXT',
   'PARAGRAPH_TYPE_CODE', 'PARAGRAPH_TYPE_SYNTAX_ERROR',
-  'PYTHON_PRIMER_BUTTON_NAME', 'ERROR_COUNTER_LANGUAGE_UNFAMILIARITY',
-  'ERROR_COUNTER_SAME_RUNTIME', 'UNFAMILIARITY_THRESHOLD',
-  'FEEDBACK_CATEGORIES',
+  'PARAGRAPH_TYPE_OUTPUT', 'PYTHON_PRIMER_BUTTON_NAME',
+  'ERROR_COUNTER_LANGUAGE_UNFAMILIARITY', 'ERROR_COUNTER_SAME_RUNTIME',
+  'UNFAMILIARITY_THRESHOLD', 'FEEDBACK_CATEGORIES',
+  'TEST_SUITE_ID_SAMPLE_INPUT', 'CORRECTNESS_STATE_STARTING',
+  'CORRECTNESS_STATE_INPUT_DISPLAYED',
+  'CORRECTNESS_STATE_EXPECTED_OUTPUT_DISPLAYED',
+  'CORRECTNESS_STATE_OBSERVED_OUTPUT_DISPLAYED', 'CORRECTNESS_FEEDBACK_TEXT',
   function(
     FeedbackObjectFactory, TranscriptService, ReinforcementGeneratorService,
     CODE_EXECUTION_TIMEOUT_SECONDS, SUPPORTED_PYTHON_LIBS,
     RUNTIME_ERROR_FEEDBACK_MESSAGES, WRONG_LANGUAGE_ERRORS, LANGUAGE_PYTHON,
     CLASS_NAME_AUXILIARY_CODE, CLASS_NAME_SYSTEM_CODE, PARAGRAPH_TYPE_TEXT,
     PARAGRAPH_TYPE_CODE, PARAGRAPH_TYPE_SYNTAX_ERROR,
-    PYTHON_PRIMER_BUTTON_NAME, ERROR_COUNTER_LANGUAGE_UNFAMILIARITY,
-    ERROR_COUNTER_SAME_RUNTIME, UNFAMILIARITY_THRESHOLD,
-    FEEDBACK_CATEGORIES) {
+    PARAGRAPH_TYPE_OUTPUT, PYTHON_PRIMER_BUTTON_NAME,
+    ERROR_COUNTER_LANGUAGE_UNFAMILIARITY, ERROR_COUNTER_SAME_RUNTIME,
+    UNFAMILIARITY_THRESHOLD, FEEDBACK_CATEGORIES,
+    TEST_SUITE_ID_SAMPLE_INPUT, CORRECTNESS_STATE_STARTING,
+    CORRECTNESS_STATE_INPUT_DISPLAYED,
+    CORRECTNESS_STATE_EXPECTED_OUTPUT_DISPLAYED,
+    CORRECTNESS_STATE_OBSERVED_OUTPUT_DISPLAYED, CORRECTNESS_FEEDBACK_TEXT) {
+
+    /**
+     * Object used to keep track of which state we are in for correctness
+     * feedback.
```

Comment:
This docstring could use more detail; what are the keys and values? (In particular I think info about the keys is missing.)

Reply:
Fixed